test/src/tests/widgets/test_label.py

```diff
@@ -0,0 +1,29 @@
+import System.Windows.Forms as WinForms
+from pytest import fixture
+from tests.utils import set_get
+
+import toga
+
+TEXTS = ["", "a", "ab", "abc", "hello world"]
+
+
+# TODO: refactor duplication with other widgets
+@fixture
+def widget(main_box):
+    label = toga.Label("")
+    main_box.add(label)
+    yield label
+    main_box.remove(label)
+
+
+def test_native(widget, native, main_box):
+    native_box = main_box._impl.native
+    assert native_box.Controls.Count == 1
+    # TODO: assert native_box.Controls[0] is native
+    assert isinstance(native_box.Controls[0], WinForms.Label)
```

Comment:
The `is` check doesn't currently work, so maybe pythonnet doesn't enforce a 1-1 relationship between .NET objects and their Python proxies. I couldn't find any information about this in a quick search. @freakboy3742: any idea?

Reply:
It looks like [`Object.ReferenceEquals`](https://learn.microsoft.com/en-us/dotnet/api/system.object.referenceequals?view=net-7.0) does what we need.